components/scream/src/physics/p3/CMakeLists.txt

```diff
@@ -15,7 +15,6 @@ set(P3_SRCS
   ${SCREAM_BASE_DIR}/../eam/src/physics/cam/micro_p3.F90
```

Comment:
I'm not sure why this doesn't cause a link error.  Perhaps @jgfouca has an idea.  I'm pretty sure this is included here for the C++/F90 BFB unit tests.  I don't think any of the code in the AD interface call micro_p3.F90 directly.

Reply:
For what it's worth: if the function that needs `micro_p3_utils_init_c` isn't called itself, the linker is smart enough to eliminate the dependency.

Reply:
Ah, it appears that our `physics_share` library compiles `eam/src/physics/cam/micro_p3_utils.f90`. I am a bit puzzled on why physics_share should depend on `micro_p3_utils.f90`. It seems to me that such file, as the name implies, is p3-specific.

@singhbalwinder digging in git history, it seems this dep was added when Murphy-Koop stuff was added. Do you know if this _really_ requires micro_p3_utils, or if it can be removed from `physics_share`?

Reply:
I moved MurphyKoop in `wv_sat_scream.F90` so that both P3 and SHOC can use MurphyKoop. MurphyKoop depends on some variables which P3 initializes (zerodegc and ep_2), therefore I had that dependency. I agree that it should be removed. 

For removing this dependency, we might need to re-shuffle some codes and add codes to initialize these variables (zerodegc and ep_2) in the shared codes.